http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcRequestAuthenticator.java

```diff
@@ -336,9 +346,16 @@ protected AuthChallenge checkStateCookie() {
             log.warn("state parameter was null");
             return challenge(HttpStatus.SC_BAD_REQUEST, AuthenticationError.Reason.INVALID_STATE_COOKIE, null);
         }
-        if (!state.equals(stateCookieValue)) {
+
+        String sCookieValue = stateCookieValue;
```

Comment:
Why are these changes needed?

The `checkStateCookie()` method currently deals specifically with the `deployment.getStateCookieName()` (i.e., the `OAuth_Token_Request_State` cookie).

Reply:
I think this still needs to be addressed. The state cookie is something separate that's unrelated to the new cookie that's being introduced.

Reply:
When more than 1 cookie is set in exchange.getResponse() (see line 324-326)
they are returned in the request as a string of concantenated text in
this format: 1st-cookie-value; 2nd-cookie-name=2nd-cookie-value; .. etc.
This check is evaluating "stateCookieValue" which is part of the pre-existing
code.  I have changed the initial variable name to be more generic. I
think the original variable name should remain for consistency and clarity.

Reply:
That doesn't sound right. There should be an `ArrayList` that contains 2 separate `HttpServerCookies`, each with their own values. That's why changes should not be needed to the `String stateCookieValue = getCookieValue(deployment.getStateCookieName());` line.

Is it possible the concatentation stuff you mentioned is due to changes made to the test itself?

Reply:
okhttp3.mockwebserver.RecordedRequest appears to only return headers.  For "Cookie" it is of the format described.  I made changes in AbstractBaseHttpTest to parse this and provide an array of cookies as needed.